src/main/ank/1.10-natural-transformations.md

```diff
@@ -143,18 +143,17 @@ length [] = Const 0
 length (x:xs) = Const (1 + Const (length xs))
 ```
 ```kotlin:ank:playground
-import arrow.typeclasses.Const
-import arrow.typeclasses.fix
-import arrow.core.extensions.const.semigroup.semigroup
-import arrow.core.extensions.semigroup
+import arrow.core.Const
+import arrow.core.extensions.const.monoid.monoid
+import arrow.core.extensions.monoid
+import arrow.core.fix
 
-tailrec fun <A> List<A>.length(acc: Const<Int, A> = Const.just(0)): Const<Int, Any?> =
+fun <A> List<A>.length(): Const<Int, A> =
   when (isEmpty()) {
-    true -> acc
-    false -> drop(1).length(
-      Const.semigroup<Int, A>(Int.semigroup())
-        .run { acc + Const.just<Int, A>(1) }.fix()
-    )
+    true -> Const.just(0)
+    false -> Const.monoid<Int, A>(Int.monoid()).run {
+      Const.just<Int, A>(1) + drop(1).length()
```

Comment:
Does this compile? You're adding `Const(1) + n`, being n an integer. Shouldn't it be `Const(1 + n)`?

Reply:
It does, Ank verifies it.
This is `Const(1) + n` where n is of type `Const<Int, A>`

Reply:
`n` is the length of the tail, isn't it?

Reply:
Correct